internal/testrunner/runners/system/tester.go

```diff
@@ -1987,10 +1995,78 @@ func (r *tester) checkTransforms(ctx context.Context, config *testConfig, pkgMan
 		if err != nil {
 			return fmt.Errorf("creating fields validator for data stream failed (path: %s): %w", transformRootPath, err)
 		}
-		if errs := validateFields(transformDocs, fieldsValidator); len(errs) > 0 {
-			return testrunner.ErrTestCaseFailed{
-				Reason:  fmt.Sprintf("errors found in documents of preview for transform %s for data stream %s", transformId, dataStream),
-				Details: errs.Error(),
+		if r.fieldValidationMethod == allMethods || r.fieldValidationMethod == fieldsMethod {
+			if errs := validateFields(transformDocs, fieldsValidator); len(errs) > 0 {
+				return testrunner.ErrTestCaseFailed{
+					Reason:  fmt.Sprintf("errors found in documents of preview for transform %s for data stream %s", transformId, dataStream),
+					Details: errs.Error(),
+				}
+			}
+		}
+
+		if r.fieldValidationMethod == allMethods || r.fieldValidationMethod == mappingsMethod {
+			destIndexTransform := transform.Definition.Dest.Index
+			// Index Template format is: "<type>-<package>.<transform>-template"
+			// For instance: "logs-ti_anomali.latest_intelligence-template"
+			indexTemplateTransform := fmt.Sprintf("%s-%s.%s-template",
+				ds.Inputs[0].Streams[0].DataStream.Type,
+				pkgManifest.Name,
+				transform.Name,
+			)
+
+			// TODO: use the docs obtained here for the fields validator too
+			// In order to compare the mappings, it is required to wait until the documents has been
+			// ingested in the given transform index
+			// It looks like that not all documents ingested previously in the main data stream are going
+			// to be ingested in the destination index of the transform. That's the reason to disable
+			// the asserts
+			waitOpts := waitForDocsOptions{
+				period: 5 * time.Second,
+				// Add specific timeout to ensure transform processes the documents (depends on "delay" field?)
+				timeout: 10 * time.Minute,
+				stopCondition: func(hits *hits, oldHits int) (bool, error) {
+					deleted, err := r.getDeletedDocs(ctx, destIndexTransform)
+					if err != nil {
+						return false, err
+					}
+					foundDeleted := deleted > 0
+					// Wait at least for the number of documents found after running the transform preview
+					foundHits := hits.size() >= len(transformDocs)
```

Comment:
Why waiting for this number of documents?

Reply:
At first, I was waiting for just any document (`> 0`).

But I thought that probably there could have at least the same number of documents that the ones obtained in the preview. Sometimes, I saw that there are even more documents.

I wanted to wait for some amount of documents here, so the mappings created in the destination index could be more accurate.

Reply:
Not a strong opinion, just suggestions in case we need to make things faster.

Reply:
I'll do some tests looking at the stats :eyes:

Reply:
I used as stop condition the number of processed docs, and it looks promising.

I kept the same `waitForDocs` function to keep checking the failure store too as it is performed with the regular data streams.

I was wondering to do a specific waiting method for the processed docs, but probably it is good to keep the check of the failure store. WDYT?